pkg/operator/controllers/clusteroperatoraro/clusteroperatoraro_controller.go

```diff
@@ -0,0 +1,187 @@
+package clusteroperatoraro
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+const (
+	clusterOperatorName = "aro"
+)
+
+// The default set of status change reasons.
+const (
+	reasonAsExpected   = "AsExpected"
+	reasonInitializing = "Initializing"
+)
+
+type ClusterOperatorAROReconciler struct {
+	// Replace configcli with CO client
+	configcli configclient.Interface
+	arocli    aroclient.Interface
+	log       *logrus.Entry
+}
+
+func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, configcli configclient.Interface) *ClusterOperatorAROReconciler {
+	return &ClusterOperatorAROReconciler{
+		configcli: configcli,
+		arocli:    arocli,
+		log:       log,
+	}
+}
+
+// SetupWithManager setup our manager
+func (r *ClusterOperatorAROReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	aroCOPredicate := predicate.NewPredicateFuncs(func(meta metav1.Object, object runtime.Object) bool {
```

Comment:
Can probably even get rid of this now. I previously was using Watch instead of Owns as I did not have controller reference on the object. It now should work without the predicate.